urllib3/connectionpool.py

```diff
@@ -666,6 +666,10 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
                                 release_conn=release_conn, body_pos=body_pos,
                                 **response_kw)
 
+        def drain_and_release_conn(response):
+            response.read()
```

Comment:
Ok, so that leads us back to how to properly discard the connection without modifying the Response interface?